src/coreclr/tools/Common/TypeSystem/Ecma/EcmaSignatureTranslator.cs

```diff
@@ -179,7 +178,9 @@ public void ParseType()
                 case SignatureTypeCode.Array:
                     {
                         ParseType();
+#pragma warning disable IDE0059 // The value is never used but is informative for other people to know what is being parsed
                         var rank = ParseCompressedInt();
+#pragma warning restore IDE0059 // Unnecessary assignment of a value
```

Comment:
The recommendation is to change it to underscore since the value will not be used, I think to leave it as `rank` is more useful, given that the operation needs to be executed.

Reply:
What I usually do is to put the code that the analyzer does not like in a comment. Like `/* var rank = */ ParseCompressedInt();`